base_layer/core/src/validation/block_validators.rs

```diff
@@ -99,6 +122,77 @@ impl FullConsensusValidator {
     pub fn new(rules: ConsensusManager) -> Self {
         Self { rules }
     }
+
+    /// Calculates the achieved and target difficulties at the specified height and compares them.
+    fn check_achieved_and_target_difficulty<B: BlockchainBackend>(
```

Comment:
Why have you removed the help function?
This will be shared between block sync, pruning sync, header sync etc.

Reply:
It's self contained and will not be needed if the fast header / block sync is used.